### PR TITLE
Added the ability to save entities to the DB.

### PR DESCRIPTION
**Pull request by cesarbiods**
Testing is done through the create endpoint I added but feel free to rip it out in the future. I had to rework the DB initialization to use peewee's functions instead of the default `sqlite3`'s otherwise peewee wouldn't save the objects. It has to have the connection to persist data.

### CODE REVIEW

**Review by matthew-l-craig — APPROVED** [review]

File: main.py [hunk 1]
```diff
@@ -51,6 +50,35 @@ async def get_book_by_isbn(isbn: str) -> BookDto:
...
+        reading_level="Advanced",
+        banned_book=False,
+        topics="Adventure, Exploration",
+        subjects="Literature",
+        target_audience="Adults",
+        alternate_titles="The Example Book"
+    )
+    print(f"Book created: {new_book.title} with ID {new_book.book_id}")
+    book_entity = BookMapper.dto_to_entity(book)
+    book_entity.save()
```
Comment (rael346) [inline main.py:83]: A bit confused here. Which line is actually adding the book here, `BookEntity.create()` or `book_entity.save()`?

File: db.py [hunk 2]
```diff
@@ -1,21 +1,32 @@
 import sqlite3
+from peewee import SqliteDatabase
 import os
 
+# Initialize Peewee database
+db = SqliteDatabase('remo.db')
```
Comment (rael346) [inline db.py]: When using SQLite in general, there are some pragmas, or options in SQLite, that you will want to enable. This mainly makes SQLite a bit more consistent. See `peewee` docs for the [recommended pragmas](https://docs.peewee-orm.com/en/latest/peewee/database.html#recommended-settings)

File: db.py [hunk 3]
```diff
@@ -1,21 +1,32 @@
...
         files = os.listdir('sql')
         for file in files:
             with open(f'sql/{file}', 'r') as f:
                 query = f.read()
-                self.cur.execute(query)
+                # Use Peewee's execution method
+                self.db.execute_sql(query)
 
     def query(self, query):
-        self.cur.execute(query)
```
Comment (rael346) [inline db.py:18]: seems like these methods are unnecessary now that we are using entity to insert and read data?

**Review by rael346 — CHANGES_REQUESTED** [review]
left a few questions. Also before you do a commit, do run `uvx ruff check` for linting (unused imports, etc) and `uvx ruff format` for formating

File: main.py (hunk 1 above)
Comment (cesarbiods) [reply main.py:83]: Turns out they are both used for persisting data. The difference between them is very subtle as seen [here](https://docs.peewee-orm.com/en/latest/peewee/quickstart.html#storing-data)

File: db.py (hunk 3 above)
Comment (cesarbiods) [reply db.py:18]: This part here creates the tables. I didn't test it without it but I definitely deleted the Db instance file and had it regenerate when starting the app.

File: db.py (hunk 2 above)
Comment (cesarbiods) [reply db.py]: Thanks for the heads-up.  Incoming change

**Pull request merged**